### PR TITLE
Clean up encode_partition_topdown()

### PR DESCRIPTION
**Pull request by redzic**
Removes some unnecessary copies and usages of ArrayVec.

### CODE REVIEW

**Comment by redzic** [comment]
looks like CI is failing due to unrelated network issues

File: src/encoder.rs [hunk 1]
```diff
@@ -2933,22 +2933,22 @@ fn encode_partition_topdown<T: Pixel, W: Writer>(
       (bsize > fi.partition_range.min && is_square) || must_split
     };
 
-  let mut rdo_output =
-    block_output.clone().unwrap_or(PartitionGroupParameters {
+  let mut rdo_output = if let Some(block_output) = block_output {
+    block_output.clone()
+  } else {
+    PartitionGroupParameters {
```
Comment (barrbrain) [inline src/encoder.rs:2939]: How does using `unwrap_or_else()` compare? It should be equivalent after optimisation but more concise.
```suggestion
  let mut rdo_output =
    block_output.clone().unwrap_or_else(|| PartitionGroupParameters {
```

File: src/encoder.rs [hunk 2]
```diff
@@ -3156,7 +3156,7 @@ fn encode_partition_topdown<T: Pixel, W: Writer>(
             &Some(PartitionGroupParameters {
               rd_cost: mode.rd_cost,
               part_type: PartitionType::PARTITION_NONE,
-              part_modes: once(mode).collect(),
+              part_modes: ArrayVec::try_from(&[mode] as &[_]).unwrap(),
```
Comment (barrbrain) [inline src/encoder.rs:3159]: Is the following equivalent?
```suggestion
              part_modes: [mode][..].try_into().unwrap(),
```

**Review by barrbrain — COMMENTED** [review]
Thank you for cleaning out the cruft. :)

File: src/encoder.rs (hunk 2 above)
Comment (redzic) [reply src/encoder.rs:3159]: It should be, since `try_into` calls `try_from` anyway.

File: src/encoder.rs (hunk 1 above)
Comment (redzic) [reply src/encoder.rs:2939]: It does seem equivalent after optimization. For some reason I thought clone and then unwrap_or_else would be worse then cloning inside an if-let block, but I guess not.

File: src/encoder.rs [hunk 3]
```diff
@@ -2934,21 +2934,19 @@ fn encode_partition_topdown<T: Pixel, W: Writer>(
...
 
   let mut rdo_output =
-    block_output.clone().unwrap_or(PartitionGroupParameters {
+    block_output.clone().unwrap_or_else(|| PartitionGroupParameters {
       part_type: PartitionType::PARTITION_INVALID,
       rd_cost: std::f64::MAX,
       part_modes: ArrayVec::new(),
-    });
+    }
+  };
```
Comment (barrbrain) [inline src/encoder.rs:2942]: Oops, my suggestion didn't include reverting this hunk.
```suggestion
  });
```

**Review by barrbrain — APPROVED** [review]

**Pull request merged**